### PR TITLE
Expose only public APIs in inproc RPC client

### PR DESCRIPTION
**Pull request by adambabik**
Currently, via `CallRPC` binding which is used as a provider for web3.js, we expose all registered APIs, even private ones. This PR makes sure that we expose only public APIs.

Todo:
- [x] go-ethereum patch is missing.

### CODE REVIEW

**Comment by dshulyak** [comment]
i think we need to provide access to this API in some way (sign/ecrecover)

```
 {
			Namespace: "personal",
			Version:   "1.0",
			Service:   NewPrivateAccountAPI(apiBackend, nonceLock),
			Public:    false,
		},
```

should we extract those two methods into public api that will use our confirmation backend?

cc @mandrigin @flexsurfer

**Comment by adambabik** [comment]
I changed the implementation to take into consideration `APIModules` value from `params.NodeConfig` so `personal` can be whitelisted (and it currently is).

As we discussed, we probably don't want to expose all methods from `personal` module even if whitelisted, so the possible solution is to remove it from whitelisted modules and provide only selected methods.

And the final note, we may have `APIModules` to be configurable from the app through Profile tab and Advanced Settings.

File: geth/api/backend.go [hunk 1]
```diff
@@ -109,7 +111,14 @@ func (b *StatusBackend) startNode(config *params.NodeConfig) (err error) {
 			err = fmt.Errorf("node crashed on start: %v", err)
 		}
 	}()
-	err = b.statusNode.Start(config)
+
+	services := []gethnode.ServiceConstructor{
+		func(_ *gethnode.ServiceContext) (gethnode.Service, error) {
+			return mailservice.New(b.statusNode), nil
```
Comment (dshulyak) [inline geth/api/backend.go]: isn't it better to add mailservice in the same place as shhext?

**Review by dshulyak — APPROVED** [review]

File: geth/api/backend.go (hunk 1 above)
Comment (adambabik) [reply geth/api/backend.go]: Yeah, it should be more consistent.

**Comment by adambabik** [comment]
@dshulyak can you please take one more look? Thanks to your suggestion I managed to remove MailService mock and create a successful unit test for `RequestMessages` in `mailservice` package.

File: geth/mailservice/mailservice_api_test.go [hunk 2]
```diff
@@ -18,39 +19,45 @@ func TestRequestMessagesDefaults(t *testing.T) {
...
+func TestRequestMessages(t *testing.T) {
+	var err error
+
 	shh := whisper.New(nil)
-	// Node is ephemeral (only in memory).
-	nodeA, nodeErr := node.New(&node.Config{NoUSB: true})
-	require.NoError(t, nodeErr)
-	require.NoError(t, nodeA.Start())
+	aNode, err := node.New(&node.Config{
+		NoUSB: true,
```
Comment (dshulyak) [inline geth/mailservice/mailservice_api_test.go]: does it matter if it is false?

File: geth/mailservice/mailservice.go [hunk 3]
```diff
@@ -54,3 +51,7 @@ func (s *MailService) Start(server *p2p.Server) error {
 func (s *MailService) Stop() error {
 	return nil
 }
+
+func (s *MailService) nodeID() *ecdsa.PrivateKey {
+	return s.node.Server().PrivateKey
```
Comment (dshulyak) [inline geth/mailservice/mailservice.go]: it is also possible to use server that is sent to Start and remove node dependency

**Review by dshulyak — APPROVED** [review]

File: geth/mailservice/mailservice_api_test.go (hunk 2 above)
Comment (adambabik) [reply geth/mailservice/mailservice_api_test.go]: I remember that in geth 1.7.x it was throwing an exception but you didn't specify in your PR so I guess I can remove it as well.

**Comment by themue** [comment]
Just stumbled upon, as a small refactoring step while you're in package `mailserver`: filename `mailservice_api.go` sounds redundant (a bit like mentioned for package content here: https://blog.golang.org/package-names), `api.go` would be enough.

But not important.

File: geth/mailservice/mailservice_api_test.go [hunk 4]
```diff
@@ -60,19 +67,44 @@ func TestRequestMessagesFailures(t *testing.T) {
...
 	require.Contains(t, err.Error(), "Could not find peer with ID")
 	require.False(t, result)
-}
 
-func TestRequestMessagesSuccess(t *testing.T) {
-	// TODO(adam): next step would be to run a successful test, however,
-	// it requires to set up emepheral nodes that can discover each other
-	// without syncing blockchain. It requires a bit research how to do that.
-	t.Skip()
+	// with a peer acting line a mailserver
```
Comment (themue) [inline geth/mailservice/mailservice_api_test.go]: "acting line" or "acting like"?

File: geth/node/node.go [hunk 5]
```diff
@@ -187,25 +185,44 @@ func activateShhService(stack *node.Node, config *params.NodeConfig) error {
 
 			var mailServer mailserver.WMailServer
 			whisperService.RegisterServer(&mailServer)
-			mailServer.Init(whisperService, whisperConfig.DataDir, whisperConfig.Password, whisperConfig.MinimumPoW)
+			mailServer.Init(whisperService, config.WhisperConfig.DataDir, config.WhisperConfig.Password, config.WhisperConfig.MinimumPoW)
```
Comment (themue) [inline geth/node/node.go]: Looks worth a multi-line function call now.

**Review by themue — COMMENTED** [review]
Looks mostly fine so far but need a second scan.

File: geth/mailservice/mailservice_api_test.go (hunk 4 above)
Comment (adambabik) [reply geth/mailservice/mailservice_api_test.go]: Should be `acting as` :D

**Review by mandrigin — APPROVED** [review]
LGTM

**Review by themue — APPROVED** [review]
LGTM

**Pull request merged**